### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by Bombe** [comment]
You need to stop this.

**Comment by xor-freenet** [comment]
@Bombe: Thank you for offering your help with maintaining Freetalk pull requests.
Fortunately I never stepped down as maintainer of Freetalk so you do not need to deal with any Freetalk PRs whatsoever. And for the sake having a coherent procedure and especially codebase I would be thankful if you could in fact not do it even as a voluntary service. "Too many cooks spoil the broth". Thanks nevertheless though!

@wlisik: Thanks, I will look at this the following days!

**Comment by wlisik** [comment]
@xor-freenet No hurry with this. Thanks for reply.

**Review by xor-freenet — COMMENTED** [review]
General feedback:
When doing changes for which the reason is not completely apparent the commit message should tell the reason.
Tell us not only _what_ you're changing but also _why_.

Thanks :)

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1 @@
-build
-build-test
-dist
+
```
Comment (xor-freenet) [inline .gitignore]: Why are you removing these from Git's ignore list?
They are temporary directories which contain the compiler output. So as they're temporary local stuff they shouldn't be committed to the repository. Knowing that, do you still have any reason to remove them?

**Comment by xor-freenet** [comment]
Starting from the second commit 06f394a10736daa8e966c8245a269e94a0a9d04e the commits are way too large for review, sorry.
A commit should please contain a single "atomic unit of work". In other words: A commit should be about one change and only one. Unrelated things should be in different commits.

I would please like to ask you to re-commit your work in smaller chunks. While doing that please also make them include explanations for their reason in the commit messages.
You can re-commit like this:
```bash
# Check out your original code into a new branch.
git checkout -b code-cleanup-new 6d2e4df6ed26d1d8a37564fa998a76263d1bd23c 
# Forget the commits but keep the changes to the files on disk
git reset origin/master
# Commit the changes in small chunks. Repeat as often as you need to. 
# Ideally, if you use GPG, also add the parameter "--gpg-sign"
git commit --patch
# In case you notice that you forgot something in the previous commit and want to put into it
# you can use the following.
# Please don't use this after you did a "git push"! And: This also works with --gpg-sign.
git commit --patch --amend
# Notice: --patch only applies to changes to existing files.
# To add your new files, use the following procedure:
# Step 1: List new files:
git ls-files --others --exclude-standard
# Step 2: Commit new files ("git commit" also accepts --gpg-sign):
git add filename
git commit
# Push your changes to Github:
git push origin code-cleanup-new
# Then file a new pull request on the GitHub website
```

Thank you :)

**Comment by xor-freenet** [comment]
Closing: The submitter has deleted their user account, and the code quality of this isn't in a state where it seems worth trying to fix it.

**Pull request closed without merging**